openehr-rm/src/main/java/com/nedap/archie/rm/archetyped/Pathable.java

```diff
@@ -105,13 +111,30 @@ protected String getParentAttributeName() {
     /**
      * Path from the toplevel-RM object. Not sure if this should be here, because the EHR and Folder objects are also in
      * the RM. But for now, it works because the most toplevel element is a Composition
-     *
+     * <p>
      * API subject to change in the future!
+     *
      * @return
      */
     @RMPropertyIgnore
     public final String getPath() {
         return PathUtil.getPath(getPathSegments());
     }
 
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) return true;
+        if (o == null || getClass() != o.getClass()) return false;
+
+        Pathable pathable = (Pathable) o;
+
+        if (parent != null ? !parent.equals(pathable.parent) : pathable.parent != null) return false;
```

Comment:
I think it's NOT a good idea to check parent equals here. It will lead to endless loops, as `parent.equals` will call `this.equals`, which will call `parent.equals` etc...